src/TropicalGeometry/semiring_map.jl

```diff
@@ -29,13 +74,16 @@ end
 #
 ################################################################################
 valued_field(nu::TropicalSemiringMap) = nu.valued_field
-uniformizer_field(nu::TropicalSemiringMap) = nu.uniformizer_field
+uniformizer_in_field(nu::TropicalSemiringMap) = nu.uniformizer_in_field
 valued_ring(nu::TropicalSemiringMap) = nu.valued_ring
-uniformizer_ring(nu::TropicalSemiringMap) = nu.uniformizer_ring
-uniformizer(nu::TropicalSemiringMap) = uniformizer_ring(nu)
+uniformizer_in_ring(nu::TropicalSemiringMap) = nu.uniformizer_in_ring
 residue_field(nu::TropicalSemiringMap) = nu.residue_field
 tropical_semiring(nu::TropicalSemiringMap) = nu.tropical_semiring
 
+domain(nu::TropicalSemiringMap) = isnothing(valued_field(nu)) ? valued_ring(nu) : valued_field(nu)
```

Comment:
```suggestion
domain(nu::TropicalSemiringMap{DomainType}) where {DomainType} = (isnothing(valued_field(nu)) ? valued_ring(nu) : valued_field(nu))::DomainType
```
Maybe add those (and similar) in the next round of improvements.

Reply:
(Otherwise it is not type stable.)